stdlib/gc.mli

```diff
@@ -143,7 +145,9 @@ type control =
        Default: 0. *)
 
     max_overhead : int;
-    (** Heap compaction is triggered when the estimated amount
+    (** Note: As of OCaml 5.0, compaction is not implemented, and this field
+       will have no effect.
+       Heap compaction is triggered when the estimated amount
```

Comment:
For the control words, I think this ought to be stronger - i.e. "must be zero" and `Gc.set` should raise `Invalid_argument` if it's not zero.

Reply:
It seems to me that raising `Invalid_argument` is a bit strong?
Isn't there a middle ground where we could notify users this is not taken into account without breaking their programs?

Reply:
My rationale is that these functions should be being called correctly - i.e. that it's a programming error to call `Gc.set` with an invalid set of parameters for a given version of OCaml (i.e. the user's program _is_ broken, whether or not we raise an exception!). It doesn't seem right to me to be silently dropping the incorrect configuration request, at which point the only option is an exception and `Invalid_argument` seems more appropriate here than `Failure`?